pytorch_lightning/trainer/trainer.py

```diff
@@ -941,7 +941,8 @@ def _pre_training_routine(self):
 
         # print model summary
         if self.is_global_zero and self.weights_summary is not None and not self.testing:
-            ref_model.summarize(mode=self.weights_summary)
+            max_depth = ModelSummary.MODES[self.weights_summary]
```

Comment:
Should the Trainer.weights_summary typing signature be updated to allow int?

And should we deprecate passing a str for it?

Reply:
I did not make the changes to `Trainer.weights_summary` following https://github.com/PyTorchLightning/pytorch-lightning/pull/8062#discussion_r655532083 . But it might be less confusing to change everything to an int based argument at once, making it consistent.

Let me know what you prefer.